libcontainerd/client_daemon.go

```diff
@@ -284,10 +283,6 @@ func (c *client) Exec(ctx context.Context, containerID, processID string, spec *
 	})
 	if err != nil {
 		close(stdinCloseSync)
-		if rio != nil {
-			rio.Cancel()
-			rio.Close()
-		}
```

Comment:
This was duplicate code, there is already a `defer if err != nil` above which calls cancel/close.